### ISSUE TITLE
Error on first run

### ISSUE THREAD

**Issue by nbigaouette** [sectore/timr#30]
Trying this out, I get an error on the first run:

```
❯ timr
Error: Failed to get state directory

Location:
    src/config.rs:33:24
```

This is on 88293898f6ece7ccc212aa7b23a4de0e. Note that I "installed" it through cargo (but it probably shouldn't affect the error):
```
❯ cargo install --path=.
```

**Comment by nbigaouette** [comment]
Debugging this a little bit, I have those values:
```
[src/config.rs:30:31] get_project_dir()? = ProjectDirs {
    project_path: "timr",
    cache_dir: "/Users/username/Library/Caches/timr",
    config_dir: "/Users/username/Library/Application Support/timr",
    config_local_dir: "/Users/username/Library/Application Support/timr",
    data_dir: "/Users/username/Library/Application Support/timr",
    data_local_dir: "/Users/username/Library/Application Support/timr",
    preference_dir: "/Users/username/Library/Preferences/timr",
    runtime_dir: None,
    state_dir: None,
}
[src/config.rs:30:26] dbg!(get_project_dir()?).state_dir() = None
[src/config.rs:30:21] dbg!(dbg!(get_project_dir()?).state_dir()).map(|d| d.to_path_buf()) = None
```
Note that I am on macOS. So `state_dir` is `None`, and thus [the call to](https://github.com/sectore/timr/blob/main/src/config.rs?8829389#L31) `state_dir()` returns `None`.

According to the docs (https://docs.rs/directories/latest/directories/struct.ProjectDirs.html#method.state_dir) there is no state dir on macOS or Windows.

I guess to fix the state dir should be set to something explicit on those platforms, for example inside the config dir (not sure if that's really what would be suggested for `directories` crate).

**Comment by sectore** [comment]
@nbigaouette Thanks for reporting this issue and digging into it!

Could you check out branch [`fix/30-state-dir` ](https://github.com/sectore/timr/tree/fix/30-state-dir) as part of PR #31 to run the fix on your machine, pls? 

_Unfortunately I can't test it on my machine (Linux), but would be happy to get your feedback._

**Comment by nbigaouette** [comment]
That seems to do the trick! Thanks for fixing this :)

Will continue to explore the tool. Thanks!
